perf/perf_uprobe.py

```diff
@@ -0,0 +1,126 @@
+                        os.path.join(self.teststmpdir, 'Makefile'))
+
+        build.make(self.teststmpdir)
+        os.chdir(self.teststmpdir)
+        self.temp_file = tempfile.NamedTemporaryFile().name
+        self.cmdProbe = "perf probe -x"
+        self.recProbe = "perf record -o %s -e probe_uprobe_test:doit" % self.temp_file
+        self.report = "perf report --input=/tmp/test.out"
+        self.distro_version = detected_distro.version
+
+    def cmd_verify(self, cmd):
+        return process.run(cmd, shell=True)
+
+    def test_uprobe(self):
+        output = self.cmd_verify(self.cmdProbe + ' /usr/bin/perf main')
+        if 'Added new event' not in output.stderr:
+            self.fail("perf probe of perf main failed")
+        output = self.cmd_verify('perf probe -l')
+        if 'probe_perf:main' not in output.stdout:
+            self.fail("perf probe of 'perf main' not found in list")
+        output = self.cmd_verify('cat /sys/kernel/debug/tracing/uprobe_events')
+        if 'probe_perf' not in output.stdout:
+            self.fail("sysfs file didn't reflect uprobe events")
+        output = self.cmd_verify('perf record -o %s -e probe_perf:main -- '
+                                 'perf list' % self.temp_file)
+        if 'samples' not in output.stderr:
+            self.fail("perf.data file not created")
+        output = self.cmd_verify(self.report)
+        print(output)
+
+    def test_uprobe_return(self):
+        output = self.cmd_verify(self.cmdProbe + ' ./uprobe_test doit%return')
+        if 'Added new event' not in output.stderr:
+            self.fail("perf probe of uprobe return failed")
+        # RHEL8 onwards
+        if self.distro_name == "rhel" and self.distro_version >= '8':
+            output = self.cmd_verify(self.recProbe + '__return -- ./uprobe_test')
+        else:
+            output = self.cmd_verify(self.recProbe + ' ar ./uprobe_test')
+        if 'samples' not in output.stderr:
+            self.fail("perf.data file not created")
+        output = self.cmd_verify(self.report)
+        print(output)
+
+    def test_uretprobe_variable(self):
+        output = self.cmd_verify(self.cmdProbe + ' ./uprobe_test "doit i"')
+        if 'Added new event' not in output.stderr:
+            self.fail("perf probe of uprobe variable failed")
+        # RHEL8 onwards
+        if self.distro_name == "rhel" and self.distro_version >= '8':
+            output = self.cmd_verify(self.recProbe + ' -- ./uprobe_test')
+            print(output)
+        else:
+            output = self.cmd_verify(self.recProbe + ' ar ./uprobe_test')
+            print(output)
+        if 'samples' not in output.stderr:
+            self.fail("perf.data file not created")
+        output = self.cmd_verify(self.report)
+        print(output)
+
+    def tearDown(self):
+        output = self.cmd_verify('perf probe -d \\*')
+        if os.path.isfile(self.temp_file):
+            os.remove(self.temp_file)
```

Comment:
can this be handle it try expect or use avcoado utility process but try expect will be good option as it os command chance of failing this command is there on that case script will error out